### PR TITLE
Cherry-pick #7991 to 6.x: Add tag "truncated" to "log.flags" if incoming line is longer than configured limit

### LINKED ISSUE #7022: Tag truncated events by max_bytes
While [max_bytes](https://www.elastic.co/guide/en/beats/filebeat/6.2/configuration-filebeat-options.html#_literal_max_bytes_literal) is a useful feature to protect against unexpected large messages, it does its job silently today.  There are no logging messages indicating a message being truncated, and the resulting published event also has no tagging indicating that the event has been truncated.  If a user looks at the resulting event, they may think that the truncated version is really the original message.  It will be helpful for us to add a tag to all the events that have been truncated via max_bytes so it is easier from an auditing point of view.

### PR DESCRIPTION
**Pull request by kvch**
Cherry-pick of PR #7991 to 6.x branch. Original message: 

A new field is added to store the flags of an event named `"log.flags"`. 
If a message is truncated, "truncated" flag is added to the list.

Example event with `"truncated"` flag:
```
{
  "@timestamp": "2018-08-16T13:00:46.759Z",
  "@metadata": {
    "beat": "filebeat",
    "type": "doc",
    "version": "7.0.0-alpha1"
  },
  "host": {
    "name": "sleipnir"
  },
  "source": "/home/n/test.log",
  "offset": 33,
  "log": {
    "flags": [
       "truncated"
    ],
  },
  "message": "test line",
  "prospector": {
    "type": "log"
  },
  "input": {
    "type": "log"
  },
  "beat": {
    "hostname": "sleipnir",
    "version": "7.0.0-alpha1",
    "name": "sleipnir"
  }
}

```

Blocks #7997 
Closes #7022

### CODE REVIEW

**Comment by kvch** [comment]
It is blocked by #8166

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -79,6 +79,12 @@ https://github.com/elastic/beats/compare/v6.4.0...6.x[Check the HEAD diff]
 
 *Filebeat*
 
+- Make inputsource generic taking bufio.SplitFunc as input {pull}7746[7746]
+- Add custom unpack to log hints config to avoid env resolution {pull}7710[7710]
+- Keep raw user agent information after parsing as user_agent_raw in Filebeat modules. {pull}7823[7832]
+- Make docker input check if container strings are empty {pull}7960[7960]
+- Add tag "truncated" to "log.flags" if incoming line is longer than configured limit. {pull}7991[7991]
+
```
Comment (urso) [inline CHANGELOG.asciidoc]: Can you double check these entries?

**Comment by kvch** [comment]
jenkins test this

**Comment by kvch** [comment]
Failing tests are unrelated.

**Pull request merged**